features/accommodations/edit/AccommodationEditPage.tsx

```diff
@@ -0,0 +1,285 @@
+import { extractErrorMessage } from "../../../core/utils/errors";
+import { deleteAccommodation, deleteFolder } from "../accommodationActions";
+import { useRouter } from "next/router";
+import { PencilSquareIcon, TrashIcon } from "@heroicons/react/24/outline";
+import { bookingAdvancePeriodLabels } from "../creation/CalendarForm";
+import { toText } from "../../../core/utils/cron";
+
+export type AccommodationUpdateFormProps = {
+  accommodation: Accommodation;
+  onCancel: () => void;
+  onSuccess: (accommodation: Accommodation, photoUrls?: string[]) => void;
+  onError: (e: any) => void;
+  photoUrls?: string[];
+};
+
+type UpdateFormName =
+  | "rooms"
+  | "property"
+  | "title"
+  | "houseRules"
+  | "amenities"
+  | "location"
+  | "pricing"
+  | "photos"
+  | "availability"
+  | "calendar";
+
+const forms = {
+  rooms: dynamic(() => import("./RoomsUpdateForm")),
+  property: dynamic(() => import("./PropertyUpdateForm")),
+  title: dynamic(() => import("./TitleAndDescriptionUpdateForm")),
+  houseRules: dynamic(() => import("./HouseRulesUpdateForm")),
+  amenities: dynamic(() => import("./AmenitiesUpdateForm")),
+  location: dynamic(() => import("./LocationUpdateForm")),
+  pricing: dynamic(() => import("./PricingUpdateForm")),
+  photos: dynamic(() => import("./PhotosUpdateForm")),
+  availability: dynamic(() => import("./AvailabilityUpdateForm")),
+  calendar: dynamic(() => import("./AvailabilityCalendarUpdateForm")),
+} as const;
+
+export const AccommodationEditPage: FC<{ id: string }> = ({ id }) => {
+  const { setResult, result } = useResult("accommodations");
+  const notifications = useNotifications();
+  const [updatedAccommodation, setUpdatedAccommodation] = useState<Accommodation>();
+  const [updatedUrls, setUpdatedUrls] = useState<string[]>();
+  const { accommodation } = useAccommodation(id, [], updatedAccommodation);
+  const { photoUrls } = usePhotos(id, [], updatedUrls);
+  const router = useRouter();
+
+  const [ActiveForm, setActiveForm] = useState<ComponentType<AccommodationUpdateFormProps>>();
+
+  const setActiveModal = (name?: UpdateFormName) => {
+    setActiveForm(name ? forms[name] : undefined);
+  };
+
+  const onSuccess = (accommodation: Accommodation, photoUrls?: string[]) => {
+    notifications.success("You have successfully updated the accommodation.");
+    setResult({ status: ResultStatus.Ok, type: "UPDATE_ACCOMMODATION" });
+    setUpdatedAccommodation(accommodation);
+    if (photoUrls) setUpdatedUrls(photoUrls);
+    setActiveModal(undefined);
+  };
+
+  const onError = (error: any) => {
+    notifications.error(extractErrorMessage(error));
+    setResult({ status: ResultStatus.Error, type: "UPDATE_ACCOMMODATION" });
+    setActiveModal(undefined);
+  };
+
+  const deleteAccommodationAndPhotos = async () => {
+    if (!accommodation) return;
+    if (!confirm("Are you sure?")) return;
+    try {
+      await deleteFolder(accommodation.id);
+      await deleteAccommodation(accommodation.id);
```

Comment:
Maybe you should deleteAccommodation first. In case that fails, you will have deleted images.

Reply:
The only problem is that the delete folder operation requires an existing accommodation when authorizing. We can change it to allow deletion even when it is missing.

Reply:
Leave it as it is. The best solution would be to publish some event from DeleteHandler such as AccomodationDeletedEvent to some queue (RabbitMQ) and image service to remove them. Too much pain atm.